libraries/windowsfeature_helper.rb

```diff
@@ -0,0 +1,42 @@
+#
+# Cookbook:: iis
+# Library:: windowsfeature_helper
+#
+# Copyright:: 2017-2021, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module IISCookbook
+  module WindowsFeatureHelper
+    def transform_feature_name(install_method, names)
```

Comment:
the `install_method` is tightly coupled to the implementation of the rest of your code so this shouldn't be exposed as a public API in the recipe class.

in fact since its only used in one place it isn't really appropriate to extract it as a helper at all, so it should live with the code that calls it in the resource in a action_class declaration.

Reply:
it is also pretty weird to have the helper take the install_method and then return the names arg unmatched.

i'd say this should probably be renamed `powershell_feature_name(names)` and the check moved to the caller.

then at some later date if this class actually grows the need for multiple differen kinds of feature_name filters then you can write a method with this signature and switch/case on the install_method to transform them.